### DISCUSSION TITLE
cdc2.0.0版本依然报权限不足问题

### DISCUSSION THREAD

**Discussion by whoIsJackson** [apache/flink-cdc#408] [Q&A] [answered]
背景：cdc1.x需要reload权限，升级到了2.0.0
在本地测试时，依然报如下错误
Caused by: io.debezium.DebeziumException: User does not have the 'LOCK TABLES' privilege required to obtain a consistent snapshot by preventing concurrent writes to tables.

pom依赖如下
flink-java ->1.12.0
flink-streaming-java_2.12 -> 1.12.0
flink-clients_2.12 ->1.12.0
hadoop-client ->3.1.3
flink-connector-mysql-cdc -> 2.0.0
flink-table-planner-blink_2.12 -> 1.12.0

测试case如下
SourceFunction<String> sourceFunction = MySqlSource.<String>builder()
                .hostname("xxx")
                .port(3306)
                .username("flink_cdc2")
                .password("xxxx")
                .databaseList("xxx")
                .deserializer(new StringDebeziumDeserializationSchema())
                .serverId(5401)
                .build();
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env
                .addSource(sourceFunction)
                .print().setParallelism(1); 
        env.execute();

**Comment by whoIsJackson** [comment]
不科学，，，，，

**Comment by leonardBang** [comment, marked as answer]
MySqlSource 是 1.x 的 DataStream API，是需要锁的。
2.0 提供的无锁能力是通过SQL提供，DataStream API还未开放，如果感兴趣可以研究下 MySqlParallelSource 代码，这是2.0的 DataStream API。计划在 2.1 开放出来
